src/kv/untyped_map.h

```diff
@@ -259,7 +261,7 @@ namespace kv::untyped
           }
         }
 
-        if (changes)
+        if (changes || keep_all_writes)
```

Comment:
I don't understand this `keep_all_writes` flag. If there were any writes or removals of existing keys, then `changes` would already be set to `true` by lines 249 or 258. Why do we want to insert anything into the roll if neither of those cases has happend?

Reply:
From my understanding, when creating an untyped map for an index update the map doesn't contain the full state up to the current point, it is intended only to store the diff of what this transaction did. In this situation there might be a remove operation that occurred but won't cause the `changes = true` section as it is seen to delete an already missing key. This instead forces the logic to track the case when a delete on a missing key occurs.

With this intuition I'll move the check up earlier to just set the changes on that case.